codecs/avif/Makefile

```diff
@@ -0,0 +1,76 @@
+ROOT_DIR = node_modules
+
+CODEC_DIR = $(ROOT_DIR)/libavif
+CODEC_BUILD_DIR = build
+CODEC_OUT := $(CODEC_DIR)/$(CODEC_BUILD_DIR)/libavif.a 
+
+CODEC_URL = "https://github.com/AOMediaCodec/libavif"
+CODEC_VERSION = "v0.8.0"
+
+LIBAOM_DIR := $(CODEC_DIR)/ext/aom
+LIBAOM_BUILD_DIR = build.libavif
+LIBAOM_OUT := $(LIBAOM_DIR)/$(LIBAOM_BUILD_DIR)/libaom.a
+
+LIBAOM_URL = "https://aomedia.googlesource.com/aom/"
+LIBAOM_VERSION = "v2.0.0"
+
+OUT_JS = enc/avif_enc.js dec/avif_dec.js
+OUT_WASM = $(OUT_JS:.js=.wasm)
+
+.PHONY: all clean
+
+all: $(OUT_JS)
+
+%.js: %.cpp $(LIBAOM_OUT) $(CODEC_OUT)
+	$(CXX) \
+		-I $(CODEC_DIR)/include \
+		${CXXFLAGS} \
+		${LDFLAGS} \
+		--bind \
+		--closure 1 \
+		-s ALLOW_MEMORY_GROWTH=1 \
+		-s MODULARIZE=1 \ -s 'EXPORT_NAME="$(basename $(@F))"' \
+		-o $@ \
+		$+
+
+$(LIBAOM_OUT): $(LIBAOM_DIR)/README.md
+	mkdir -p $(LIBAOM_DIR)/$(LIBAOM_BUILD_DIR) && \
+	cd $(LIBAOM_DIR)/$(LIBAOM_BUILD_DIR) && \
+	emcmake cmake \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DENABLE_CCACHE=0 \
+		-DAOM_TARGET_CPU=generic \
+		-DENABLE_DOCS=0 \
+		-DENABLE_TESTS=0 \
+		-DCONFIG_ACCOUNTING=1 \
+		-DCONFIG_INSPECTION=0 \
+		-DCONFIG_MULTITHREAD=0 \
+		-DCONFIG_RUNTIME_CPU_DETECT=0 \
+		-DCONFIG_WEBM_IO=0 \
+		../ && \
+	$(MAKE) 
+
+$(LIBAOM_DIR)/README.md: $(CODEC_DIR)/README.md
+	mkdir -p $(LIBAOM_DIR) && \
```

Comment:
Nitpick: this doesn't have to be part of the `&&`, it works as a separate command.

Perhaps worth copy-pasting https://github.com/GoogleChromeLabs/squoosh/blob/87955ab9a03401211ef37964cdc60e95a9a96dd9/codecs/webp/Makefile#L50-L52 for consistency between Makefiles? It's also a bit simpler in that output folder is specified directly in `tar` command.